src/server.rs

```diff
@@ -0,0 +1,33 @@
+use serde::{Deserialize, Serialize};
+use std::sync::Mutex;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct JointNamesAndPositions {
+    pub names: Vec<String>,
+    pub positions: Vec<f32>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct RobotOrigin {
+    pub position: [f32; 3],
+    pub quaternion: [f32; 4],
+}
+
+#[derive(Debug)]
+pub struct ServerData {
```

Comment:
`ServerData` sounds like some configurations for the webserver for me.
How about using `RobotState`, `RobotStateData` or `RobotServerData` ...?

Reply:
`ServerRobotData`?

Reply:
From the user's point of view, I think `RobotStateHandle` would describe its nature exactly.

Reply:
I see, great!

Reply:
renamed to RobotStateHandle